src/main/java/net/cardosi/mojo/WatchMojo.java

```diff
@@ -204,16 +207,25 @@ public void execute() throws MojoExecutionException, MojoFailureException {
         }
         diskCache.release();
 
-        for (CachedProject app : projects.values()) {
-            //TODO instead of N threads per project, combine threads?
+        for (Future f : futures) {
             try {
-                app.watch();
-            } catch (IOException ex) {
-                ex.printStackTrace();
-                //TODO fall back to polling or another strategy
+                f.get();
+            } catch (ExecutionException | InterruptedException e) {
+                e.printStackTrace();
             }
         }
 
+        try {
+            // sleep 500ms, so the printouts aren't muddled
+            Thread.sleep(500);
```

Comment:
do you want this commented out? or leave for now? I know it can be improved later, once there are buffers, but it seems the least of evils, compared to unordered text.

Reply:
I have commented this out for now.